packages/bridge-ui/src/utils/getIsMintedWithEstimation.ts

```diff
@@ -0,0 +1,53 @@
+import { BigNumber, Contract, type Signer } from 'ethers';
+
+import { freeMintErc20ABI } from '../constants/abi';
+import type { Token } from '../domain/token';
+import { getLogger } from './logger';
+
+const log = getLogger('util:minting');
+
+/**
+ * This function returns a boolean indicating whether the user has already claimed
+ * the test token on L1, and the estimated cost of the transaction for minting if not.
+ */
+export async function getIsMintedWithEstimation(
+  signer: Signer,
+  token: Token,
+): Promise<[boolean, BigNumber]> {
```

Comment:
is it common to return arrays like this, or a return struct like

{
    isMinted: true,
    estimation: BigNumber
}
?

Reply:
Changed to `{ isMinted, estimatedGas }`